metrics_utility/management/validation.py

```diff
@@ -58,3 +62,61 @@ def handle_crc_ship_target(ship_target):
         billing_provider_params['red_hat_org_id'] = red_hat_org_id
 
     return billing_provider_params
+
+
+def handle_validate_date_param(
+    self,
+    options,
+):
+    params = ['since', 'until']
+    if options.get('ephemeral') and self.help_ephemeral:
+        params.append('ephemeral')
+    help_text = ''
+    exceptions = []
+    for param in params:
+        option = options.get(param)
+        if not option:
+            continue
+        if option.isdigit():
+            """If the value is a digit stop execution and render the failure message."""
+            exceptions.append('isdigit')
+            help_text = 'Integers are not allowed for parameters --since and --until.'
+            break
+        try:
+            """Try to parse the date, and if it fails go to next loop iteration.  Then, determine if we need to render the failure message."""
+            parser.parse(option)
+        except Exception:
+            help_text = getattr(self, f'help_{param}')
+            exceptions.append('parser')
+            continue
+        try:
+            """Try to parse the date, and if it fails go to next loop iteration.  Then, determine if we need to render the failure message."""
+            re.match(DAY_OR_MONTH_REGEX_PATTERN, option)
+        except Exception:
+            exceptions.append('match')
+            help_text = getattr(self, f'help_{param}')
+            continue
+    if len(exceptions) and ('isdigit' or ('parser' and 'match') in exceptions):
+        raise UnparsableParameter(help_text)
+
+
+def validate_build_extra_params(self, options):
+    opt_month = options.get('month') or None
+    since = None
+    until = None
+    handle_validate_date_param(self, options)
```

Comment:
```suggestion
    handle_validate_date_param(options.get('since'), self.help_since)
    handle_validate_date_param(options.get('until'), self.help_until)
    handle_validate_ephemeral_param(options.get('ephemeral', None), self.help_ephemeral)
```

(except ephemeral is only used for renewal guidance)

Reply:
Can we move this change `handle_validate_ephemeral_param(options.get('ephemeral', None), self.help_ephemeral)` to a separate issue?